File: top-interview-150/27.remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        j = cnt = 0
        for i in range(len(nums)):
            if nums[i] == val:
                cnt += 1
            else:
                # if not val, then put the value that current position
                nums[j] = nums[i]
                # mv to next ptr
                j += 1
        return len(nums) - cnt

File: top-interview-150/27.remove-element.kt
class Solution {
    fun removeElement(nums: IntArray, `val`: Int): Int {
        // Two Pointer approach: O(n) time, O(1) space
        var k = 0  // k equals the count of non-val values stored

        for (i in nums.indices) {
            if (nums[i] != `val`) {
                nums[k] = nums[i]
                k++
            }
        }

        return k
    }
}
